### PR TITLE
db/update! -> t2/update!

### PR DESCRIPTION
**Pull request by qnkhuat**
Convert all `db/update!` to `t2/update!`

```sh
ag db/update! .
=> no result
````
Syntaxtical changes
- changes must be a map: `(db/update User 1 :archived true) -> (t2/update! User 1 {:archived true})`

Behavior changes:
- `db/update!` returns a boolean, whereas `t2/update!` return the number of effected rows. For places we use `db/update!` as a predicate, we need to wrap it with a `pos?` to simulate the previous behavior.

Epic: https://github.com/metabase/metabase/issues/29231

### CODE REVIEW

File: modules/drivers/bigquery-cloud-sdk/test/metabase/driver/bigquery_cloud_sdk_test.clj [hunk 1]
```diff
@@ -450,7 +450,7 @@
             ;; hence, assert it was not called anymore here
             (is (= 0 @call-count) "convert-dataset-id-to-filters! should not have been called any more times"))
           ;; now, so we need to manually update the temp DB again here, to force the "old" structure
-          (let [updated? (db/update! Database db-id :details {:dataset-id "my-dataset"})]
+          (let [updated? (t2/update! Database db-id {:details {:dataset-id "my-dataset"}})]
             (is updated?)
```
Comment (tsmacdonald) [inline modules/drivers/bigquery-cloud-sdk/test/metabase/driver/bigquery_cloud_sdk_test.clj:454]: if the update fails will t2 return something falsey? I'm worried the `(is updated?)` isn't testing anything any more

Comment (qnkhuat) [reply modules/drivers/bigquery-cloud-sdk/test/metabase/driver/bigquery_cloud_sdk_test.clj:454]: yep it should be wrapped with a `pos?`. thanks for finding it.

File: modules/drivers/bigquery-cloud-sdk/test/metabase/driver/bigquery_cloud_sdk_test.clj [hunk 2]
```diff
@@ -450,7 +450,7 @@
             ;; hence, assert it was not called anymore here
             (is (= 0 @call-count) "convert-dataset-id-to-filters! should not have been called any more times"))
           ;; now, so we need to manually update the temp DB again here, to force the "old" structure
-          (let [updated? (db/update! Database db-id :details {:dataset-id "my-dataset"})]
+          (let [updated? (t2/update! Database db-id {:details {:dataset-id "my-dataset"}})]
```
Comment (qnkhuat) [inline modules/drivers/bigquery-cloud-sdk/test/metabase/driver/bigquery_cloud_sdk_test.clj:453]:
```suggestion
          (let [updated? (pos? (t2/update! Database db-id {:details {:dataset-id "my-dataset"}}))]
```

File: modules/drivers/bigquery-cloud-sdk/test/metabase/driver/bigquery_cloud_sdk_test.clj (hunk 1 above)
Comment (qnkhuat) [reply modules/drivers/bigquery-cloud-sdk/test/metabase/driver/bigquery_cloud_sdk_test.clj:454]: btw: `t2/update!` returns # of affected rows, whereas `db/update!` returns a boolean.

I had a note about this in the PR's description.

File: src/metabase/models/persisted_info.clj [hunk 3]
```diff
@@ -135,20 +135,21 @@
...
    Will ignore explicitly set `off` models."
   [database-id]
-  (db/update! PersistedInfo
-              {:where [:and
-                       [:= :database_id database-id]
-                       [:= :state "deletable"]]
-               :set {:active false,
-                     :state "creating",
-                     :state_change_at :%now}})
+  (t2/query-one
```
Comment (tsmacdonald) [inline src/metabase/models/persisted_info.clj:147]: we're sure it's just one?

File: src/metabase/sync/sync_metadata/fields/sync_metadata.clj [hunk 4]
```diff
@@ -115,7 +115,7 @@
            {:database_required new-db-required}))]
     ;; if any updates need to be done, do them and return 1 (because 1 Field was updated), otherwise return 0
     (if (and (seq updates)
-             (db/update! Field (u/the-id metabase-field) updates))
+             (not= 0 (t2/update! Field (u/the-id metabase-field) updates)))
```
Comment (tsmacdonald) [inline src/metabase/sync/sync_metadata/fields/sync_metadata.clj:118]: I'd stick with `pos?` instead of `(not= 0)`

**Review by tsmacdonald — APPROVED** [review]
image

few comments, but looks good to me. Thanks for taking this on!

File: src/metabase/models/persisted_info.clj (hunk 3 above)
Comment (qnkhuat) [reply src/metabase/models/persisted_info.clj:147]: yep. checked twice on the diff and it's the only one with this call type.

File: src/metabase/sync/sync_metadata/fields/sync_instances.clj [hunk 5]
```diff
@@ -139,7 +140,7 @@
   nested Fields. Returns `1` if a Field was marked inactive, `nil` otherwise."
   [table :- i/TableInstance, metabase-field :- common/TableMetadataFieldWithID]
   (log/info (trs "Marking Field ''{0}'' as inactive." (common/field-metadata-name-for-logging table metabase-field)))
-  (when (db/update! Field (u/the-id metabase-field) :active false)
+  (when (pos? (t2/update! Field (u/the-id metabase-field) {:active false}))
     1))
```
[the remark this answers is not in the document]
Comment (qnkhuat) [reply src/metabase/sync/sync_metadata/fields/sync_instances.clj:144]: the output is either `1 or nil`. not `1 or 0`, so the `when` check is necessary.

**Pull request merged**